src/components/table/table-provider.spec.js

```diff
@@ -265,6 +265,75 @@ describe('table > provider functions', () => {
     wrapper.destroy()
   })
 
+  it('calls provider only once when filter is object', async () => {
+    let providerCallCount = 0
+    const filter = {
+      a: '123'
+    }
+    const fields = [
+      'a' 
+    ]  
+
+    const provider = () => {
+      providerCallCount += 1
+      return testItems.slice()
+    }
+
+    mount(BTable, {
+      propsData: {
+        filter,
+        fields,
+        items: provider
+      }
+    })
+
+    await Vue.nextTick()
+
+    expect(providerCallCount).toBe(1)
+
+    await Vue.nextTick()
+
+    expect(providerCallCount).toBe(1)
+  })
+  
+  it('provider not being called when filter object changed', async () => {
+    let providerCallCount = 0
+    let lastProviderContext = null
+    const filter = {
+      a: '123'
+    }
+
+    const provider = (ctx) => {
+      providerCallCount += 1
+      lastProviderContext = ctx
+      return testItems.slice()
+    }
+
+    mount(BTable, {
+      propsData: {
+        filter,
+        items: provider
+      }
+    })
+
+    await Vue.nextTick()
+
+    expect(providerCallCount).toBe(1)
+
+    await Vue.nextTick()
+    // the provider is being called twice
+    expect(providerCallCount).toBe(2)
+
+    filter.a = '456';
```

Comment:
Just setting this will not update the prop in Vue Test Utils.  you need to call setProps on the wrapper

```js
wrapper.setProps({ filter: filter })
```